### ISSUE TITLE
Scrolling in sidebar, selected item loses focus if out of view

### ISSUE THREAD

**Issue by NiklasBr** [desktop/desktop#2957]
### Description

When you have one item selected in the item/source list for changed files and scroll the item out of view it will lose focus and not regain its focus once returned into view. See attached screen recording below.

### Version

**GitHub Desktop version:** 1.0.3

**OS version:** macOS 10.12.6 (16G29)

### Steps to Reproduce

1. Have enough changes so that Desktop will allow you to scroll.
1. Select one item.
1. Scroll so that the item is no longer visible.
1. Scroll back so that the item becomes visible again.

![2017-10-04 10 41 01](https://user-images.githubusercontent.com/279826/31167213-9531cac2-a8f1-11e7-880f-097691e3d57c.gif)

**Expected behavior:** The item should always keep its focus.

**Actual behavior:** Focus is lost.

**Reproduces how often:** 100%

### Logs

n/a

#### Additional Information

n/a

**Comment by tierninho** [comment]
Looks like we have a main and secondary color. @desktop/design can shed some light on this.

Can reproduce with clone repo list as well.

<img width="451" alt="screen shot 2017-10-04 at 10 00 37" src="https://user-images.githubusercontent.com/14828183/31167763-1728d144-a8eb-11e7-94f4-f99a650db487.png">
<img width="519" alt="screen shot 2017-10-04 at 10 00 20" src="https://user-images.githubusercontent.com/14828183/31167762-172690be-a8eb-11e7-9908-25e187104df6.png">

**Comment by NiklasBr** [comment]
It's not just the colour. When the item has focus, the focus can be moved up and down to the previous/next items. But when losing the focus (and it it using the secondary grey colour) focus can no longer be moved up or down using the arrow keys.

**Comment by shiftkey** [comment]
This is an unfortunate side-effect of using virtualized lists in many places. @niik talked a bit more about this limitation in https://github.com/desktop/desktop/issues/1669#issuecomment-314668885. I'll let him follow up with anything else, but maybe we do need to track this as a separate task.

**Comment by niik** [comment]
Thanks for opening this @NiklasBr and nice find @shiftkey. I was scouring through issues because I knew I'd written something about this before but I couldn't find it. I definitely think this is worth tracking but as I said in that comment the fix is non-trivial.

**Comment by danny007in** [comment]
can u make video,
ill try to solve

**Comment by NiklasBr** [comment]
@danny007in  video is available in the first comment in this issue as well as in #1669

**Comment by markhicken** [comment]
According to https://developer.mozilla.org/en-US/docs/Web/API/Document/activeElement#value document.activeElement is `document.body` or `null` by default or when nothing is focused. I tested this and found that Chromium does set the default value to `document.body`. I thought maybe when the selected element(s) come back into existence, we could check the `activeElement` and refocus the selected item(s) if it is still set to the body. I have submitted the following PR for review...
https://github.com/desktop/desktop/pull/15604
